doc/howtos/rdtraining/16_guidelines_pr.rst

```diff
@@ -0,0 +1,154 @@
+.. _howto/rdtraining/16_guidelines_pr:
+
+=========================================
+Part 16: Make It Shine And Submit Your PR
+=========================================
+
+In the previous chapters we saw how to create a fully functional business application. Now we will
+prepare our code, to be shared and learn how to share it.
+
+Coding guidelines
+=================
+
+We will start refactoring the code to match to the Odoo coding guidelines. The guidelines aim
+to improve the quality of Odoo Apps code.
+
+
+**Reference**: you will find the Odoo coding guidelines in :ref:`reference/guidelines`.
+
+.. exercise:: Make it shine
+
+    Refactor your code to respect the coding guidelines. Don't forget to respect the module
+    structure, the variable names, the method name convention, the model attribute order and the
+    xml ids.
+
+Your first Pull Request (PR)
+============================
+
+**The following is intended to be used by Odoo staff members only. All repositories mentioned are
+not accessible by third-party.**
+
+Now that your code respects the Odoo code guidelines and is shiny, let's share it with others.
+
+
+Configure development repository
+--------------------------------
+
+We will start creating a git development environment for the 'custom' folder, as we did it in the
+beginning for 'odoo' and 'enterprise'. The PR will target the `odoo/technical-training-sandbox` repository.
+
+To do this the first step is to
+`fork this repository in GitHub <https://guides.github.com/activities/forking/>`__ to create your
+own development repository.
+
+After you successfully created your development repository, we will configure your existing `$HOME/src/custom`
+folder to be able to commit your work.
+
+
+.. code-block:: console
+
+    $ cd $HOME/src/custom
+    $ git init
+    $ git remote add training git@github.com:odoo/technical-training-sandbox.git
+    $ git remote add training-dev git@github.com:xyz-odoo/technical-training-sandbox.git #Don't forget to change xyz-odoo to your own GitHub account
+    $ git remote set-url --push training no_push #Avoid pushing directly into the main repository
+    $ git fetch --all --prune
+    $ git branch master training/master
+    $ git checkout master #Two new files will be downloaded README.md and .gitignore
+
+Note that if your work is already in a :ref:`Git and GitHub configured folder <howto/rdtraining/02_setup/development_repository>`
+the previous steps are not needed. This is going to be the case when you will start working in the
+'odoo' or 'enterprise' folders.
+
+
+Branch, Commit & Push
+---------------------
+
+Before creating the PR, a new working branch has to be created and used to commit the code. Afterwards
+the branch will be pushed to the development repository.
+
+Create your working branch:
+
+.. code-block:: console
+
+    $ git checkout -b master-my_first_branch-xyz
```

Comment:
FWIW `--no-track` can be useful there when specifying a branch to fork explicitely e.g.
```sh
$ git checkout -b master-my-first-branch-xyz odoo/master
```
will set the upstream of the new branch to `odoo/master`, so if you just `git push` it'll try to push to master (and hopefully fail).

By setting using `git checkout --no-track ...` the upstream won't be set and `git push` will just complain that upstream is not set.